boards/hail/src/main.rs

```diff
@@ -52,6 +52,12 @@ static mut PROCESSES: [Option<&'static mut kernel::procs::Process<'static>>; NUM
     None, None, None, None,
 ];
 
+/// Dummy buffer that tricks the linker into reserving enough space for the
+/// stack.
+#[no_mangle]
+#[link_section = ".stack_buffer"]
+pub static mut STACK_MEMORY: [u8; 0x1000] = [0; 0x1000];
```

Comment:
I'm generally adverse to having the same information duplicated in two places. Is the `STACK_SIZE` variable in the linker script still used with this approach or could it possible be dropped?

Reply:
Removed.